apps/src/blockly/addons/cdoUtils.js

```diff
@@ -106,9 +109,168 @@ export function getUserTheme(themeOption) {
 }
 
 export function getCode(workspace) {
+  // Begin collecting data about our ability to correctly serialize a workspace in JSON format.
+  if (
+    experiments.isEnabled(experiments.BLOCKLY_JSON) ||
+    Math.random() <= 0.05
+  ) {
+    testJsonSerialization(workspace);
+  }
   return Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(workspace));
-  // After supporting JSON block sources, change to:
-  // return JSON.stringify(Blockly.serialization.workspaces.save(workspace));
+}
+
+function testJsonSerialization(workspace) {
+  const FIREHOSE_STUDY = 'blockly-json';
+  const FIREHOSE_EVENT_PASS = 'blocks-identical';
```

Comment:
nit/question: Is there any advantage to having these names mirror each other, something like...
```
const FIREHOSE_STUDY = 'blockly-blocks-omparison';
const FIREHOSE_EVENT_PASS = 'blockly-blocks-identical';
const FIREHOSE_EVENT_FAIL = 'blockly-blocks-different';
```
Or something? I don't have context on this PR, but I wonder if that could lead to some tiny advantage like being able to query using syntax like 
`WHERE event LIKE 'blockly-blocks-%'`.

Reply:
Since we’re not going to log successful tests, we no longer need two different events.